chassembly/src/main/java/org/quiltmc/chasm/lang/internal/parse/Parser.jj

```diff
@@ -75,10 +85,10 @@ Node literalExpression():
         { n = new FloatNode(Double.parseDouble(t.getImage())); }
         |
         t = <StringLiteral>
-        { n = new StringNode(t.getImage().substring(1, t.getImage().length() - 1)); }
+        { n = new StringNode(extractStringLiteral(t.getImage())); }
         |
         t = <CharLiteral>
-        { n = new IntegerNode((long) t.getImage().charAt(1)); }
+        { n = new IntegerNode((long) t.getImage().charAt(t.getImage().length() - 2)); }
```

Comment:
For consistency, maybe this should also use `extractStringLiteral`, so that we don't stumble over this in the future.

Reply:
Might want to rename it `extractStringOrCharLiteral` in that case